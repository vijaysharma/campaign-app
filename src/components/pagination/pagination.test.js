import React from "react";
import ReactDOM from "react-dom";
import Pagination from "./pagination";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("Pagination renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Pagination></Pagination>, div);
  ReactDOM.unmountComponentAtNode(div);
});

const currentPage = 1;
const handlePagination = jest.fn(() => currentPage++);
it("Pagination renders correctly", () => {
  const { getByTestId } = render(
    <Pagination
      campaignsPerPage={10}
      currentPage={currentPage}
      totalCampaigns={50}
      onPaginate={handlePagination}
    ></Pagination>
  );
  expect(getByTestId("pagination")).toHaveTextContent("12345");
});

it("Pagination matches snapshot", () => {
  const treeSnap = renderer
    .create(
      <Pagination
        campaignsPerPage={10}
        currentPage={currentPage}
        totalCampaigns={50}
        onPaginate={handlePagination}
      ></Pagination>
    )
    .toJSON();
  expect(treeSnap).toMatchSnapshot();
});
