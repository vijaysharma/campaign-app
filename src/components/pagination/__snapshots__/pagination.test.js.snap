// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination matches snapshot 1`] = `
<ul
  className="pagination"
  data-testid="pagination"
>
  <li
    className="page-item prev"
  >
    <a
      className="disabled"
      href="!#"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-left fa-w-8 "
        data-icon="angle-left"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link active"
      href="!#"
      onClick={[Function]}
    >
      1
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="!#"
      onClick={[Function]}
    >
      2
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="!#"
      onClick={[Function]}
    >
      3
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="!#"
      onClick={[Function]}
    >
      4
    </a>
  </li>
  <li
    className="page-item"
  >
    <a
      className="page-link"
      href="!#"
      onClick={[Function]}
    >
      5
    </a>
  </li>
  <li
    className="page-item next"
  >
    <a
      href="!#"
      onClick={[Function]}
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-right fa-w-8 "
        data-icon="angle-right"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 256 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </a>
  </li>
</ul>
`;
